Dalamud/Game/Gui/ContextMenu/ContextMenu.cs

```diff
@@ -0,0 +1,648 @@
+                handlers = handlers.Slice(7, count);
+                ids = ids.Slice(7, count);
+                for (var i = 0; i < count; ++i)
+                {
+                    if (ids[i] <= 106)
+                        continue;
+                    this.SelectedEventInterfaces.Add((nint)handlers[i].Value);
+                }
+            }
+            else
+            {
+                this.SelectedMenuType = null;
+            }
+
+            this.SubmenuItems = null;
+
+            if (this.SelectedMenuType is { } menuType)
+            {
+                lock (this.MenuItemsLock)
+                {
+                    if (this.MenuItems.TryGetValue(menuType, out var items))
+                        this.SelectedItems = new(items);
+                    else
+                        this.SelectedItems = new();
+                }
+
+                var args = new MenuOpenedArgs(this.SelectedItems.Add, this.SelectedParentAddon, this.SelectedAgent, this.SelectedMenuType.Value, this.SelectedEventInterfaces);
+                this.LogMenuOpened(args);
+                this.OnMenuOpened?.InvokeSafely(args);
+                this.SetupContextMenu(this.SelectedItems, ref valueCount, ref values);
+                Log.Verbose($"Opening {this.SelectedMenuType} context menu with {this.SelectedItems.Count} custom items.");
+            }
+            else
+            {
+                this.SelectedItems = null;
+            }
+        }
+        else if (MemoryHelper.EqualsZeroTerminatedString("AddonContextSub", (nint)addonName))
+        {
+            this.MenuCallbackIds.Clear();
+            if (this.SubmenuItems != null)
+            {
+                this.SetupContextSubMenu(this.SubmenuItems, ref valueCount, ref values);
+                Log.Verbose($"Opening {this.SelectedMenuType} submenu with {this.SubmenuItems.Count} custom items.");
+            }
+        }
+
+        var ret = this.contextAddonOpenByAgentHook.Original(module, addonName, addon, valueCount, values, agent, a7, parentAddonId);
+        if (values != oldValues)
+            this.FreeExpandedContextMenuArray(values, valueCount);
+        return ret;
+    }
+
+    private void LogMenuOpened(MenuOpenedArgs args)
+    {
+        Log.Verbose($"Got {args.MenuType} context menu with addon 0x{args.AddonPtr:X8} ({args.AddonName}) and agent 0x{args.AgentPtr:X8}");
+        if (args.Target is MenuTargetDefault targetDefault)
+        {
+            {
+                var b = new StringBuilder();
+                b.AppendLine($"Target: {targetDefault.TargetName}");
+                b.AppendLine($"Home World: {targetDefault.TargetHomeWorld.GameData?.Name.ToDalamudString() ?? "Unknown"} ({targetDefault.TargetHomeWorld.Id})");
+                b.AppendLine($"Content Id: 0x{targetDefault.TargetContentId:X8}");
+                b.AppendLine($"Object Id: 0x{targetDefault.TargetObjectId:X8}");
+                Log.Information(b.ToString());
```

Comment:
Are these intended to be `Log.Information`? If so, what is preventing logspam here?

Reply:
If so, please drop to `Verbose` or even consider removing this class entirely. How much value do these log entries have over a debugger? (n.b. that even verbose logs incur _some_ cost - this includes string interpolation.)

Reply:
Sounds good. I’ll mark this as a formal requested change then.